### PR TITLE
bunch of bug fixes, and some style changes which hopefully are reasonable.

### PR DESCRIPTION
**Pull request by fessyfoo**
TLDR; handle unicode, fetch entire index, make debug flag work, better filenames in tarballs, 
also backup dashboard metadata, allow passing of index name.

2e1ae83 (fess, 2 hours ago)
   encode dashboard files as utf8

   grafana produces unicode characters in some of it's dashboards, outputing
   unicode buffers to ascii files results in errors of the form:

   CRITICAL:root:'ascii' codec can't encode character u'\xb5' in position 15:
   ordinal not in range(128)

0730807 (fess, 2 hours ago)
   fetch all dashboards not just first 10

   use elasticsearch scan and scroll

   http://www.elasticsearch.org/guide/en/elasticsearch/guide/current/scan-scroll.html

   previously we only retrieved the default number of results which is 10 scan
   and scroll will pull back all the dashboards

a2c2661 (fess, 2 hours ago)
   make debug option work

   actually pass debug option through

2322e8f (fess, 2 hours ago)
   turn on urllib2 tracing if debug

774f334 (fess, 76 minutes ago)
   don't include tempdirs in tarball

   instead of using the entire path to the temp directory in the tarball use a
   more sensible name based of the index.  ie paths like:
   grafana-dash/dashboard_name.json

f0c963d (fess, 36 minutes ago)
   backup the rest of es results as metadata

   previously we just backed up _source.dashboard which is good for restoring
   a dashboard via the import interface of grafana.

   however if someone wanted to implement a restore direct to ES they would
   need the _id, and the rest of _source.  admittedly title and tags the
   important bits are also in the _source.dashboard there is probably value in
   the user, group and other data grafana may choose to store there.

2bb0271 (fess, 2 minutes ago)
   allow setting of index name

   Allow passing of the index name to backup. Use this name in the output
   tarball. This makes the tool usable to backup kibana dashboards from which
   grafana traces it's roots.

### CODE REVIEW

**Comment by fessyfoo** [comment]
I just realized I independently did the exact same changes as  #5 and #3 and #4
